plan/logical_plan_builder.go

```diff
@@ -195,6 +195,8 @@ func (b *planBuilder) buildJoin(join *ast.Join) LogicalPlan {
 	rightPlan := b.buildResultSetNode(join.Right)
 	newSchema := expression.MergeSchema(leftPlan.GetSchema(), rightPlan.GetSchema())
 	joinPlan := &Join{baseLogicalPlan: newBaseLogicalPlan(Jn, b.allocator)}
+	addChild(joinPlan, leftPlan)
```

Comment:
I use `attachOnConds` to construct on condition. Before this we must add all children.